White/broken pages when debug mode is enabled in multi-process environment (all Fanstatic requests yield HTTP errors)

### CKAN 2.8.3


### Please describe the expected behaviour
When debug mode is enabled (by setting [the _debug_ option](https://docs.ckan.org/en/2.8/maintaining/configuration.html#debug) to True), unless something is wrong, only a "Debug" line at the bottom and a toolbar should be added to pages.

### Please describe the actual behaviour
Pages become white and their display breaks because JavaScript and CSS resources fail to load. All requests for Fanstatic resources (such as https://example.com/recherche/fanstatic/base/:version:2020-02-14T10:39:28.07/main.js) yield a 500 error. This happens with either the event or the prefork MPM-s. The Apache log contains errors like the following:

> [Mon Mar 02 16:55:48.581011 2020] [wsgi:error] [pid 5936:tid 140678724204288] [remote 127.0.0.1:43174] mod_wsgi (pid=5936): Exception occurred processing WSGI script '/etc/ckan/default/apache.wsgi'.
> [Mon Mar 02 16:55:48.581293 2020] [wsgi:error] [pid 5936:tid 140678724204288] [remote 127.0.0.1:43174] Traceback (most recent call last):
> [Mon Mar 02 16:55:48.581339 2020] [wsgi:error] [pid 5936:tid 140678724204288] [remote 127.0.0.1:43174]   File "/usr/lib/ckan/default/lib/python2.7/site-packages/paste/registry.py", line 409, in streaming_iter
> [Mon Mar 02 16:55:48.581456 2020] [wsgi:error] [pid 5936:tid 140678724204288] [remote 127.0.0.1:43174]     for item in self.application(environ, start_response):
> [Mon Mar 02 16:55:48.581483 2020] [wsgi:error] [pid 5936:tid 140678724204288] [remote 127.0.0.1:43174]   File "/usr/lib/ckan/default/lib/python2.7/site-packages/repoze/who/middleware.py", line 86, in __call__
> [Mon Mar 02 16:55:48.581565 2020] [wsgi:error] [pid 5936:tid 140678724204288] [remote 127.0.0.1:43174]     app_iter = app(environ, wrapper.wrap_start_response)
> [Mon Mar 02 16:55:48.581605 2020] [wsgi:error] [pid 5936:tid 140678724204288] [remote 127.0.0.1:43174]   File "/usr/lib/ckan/default/lib/python2.7/site-packages/pylons/middleware.py", line 201, in __call__
> [Mon Mar 02 16:55:48.581685 2020] [wsgi:error] [pid 5936:tid 140678724204288] [remote 127.0.0.1:43174]     self.app, environ, catch_exc_info=True)
> [Mon Mar 02 16:55:48.581714 2020] [wsgi:error] [pid 5936:tid 140678724204288] [remote 127.0.0.1:43174]   File "/usr/lib/ckan/default/lib/python2.7/site-packages/pylons/util.py", line 94, in call_wsgi_application
> [Mon Mar 02 16:55:48.581780 2020] [wsgi:error] [pid 5936:tid 140678724204288] [remote 127.0.0.1:43174]     app_iter = application(environ, start_response)
> [Mon Mar 02 16:55:48.581803 2020] [wsgi:error] [pid 5936:tid 140678724204288] [remote 127.0.0.1:43174]   File "/usr/lib/ckan/default/lib/python2.7/site-packages/weberror/evalexception.py", line 251, in __call__
> [Mon Mar 02 16:55:48.581867 2020] [wsgi:error] [pid 5936:tid 140678724204288] [remote 127.0.0.1:43174]     "The EvalException middleware is not usable in a "
> [Mon Mar 02 16:55:48.581922 2020] [wsgi:error] [pid 5936:tid 140678724204288] [remote 127.0.0.1:43174] AssertionError: The EvalException middleware is not usable in a multi-process environment

### What steps can be taken to reproduce the issue? 
1. Set debug to True in CKAN's .ini file.
1. Start Apache httpd
1. Load a page (such as the homepage)

  
  
  
---
The bug seems to originate from an upstream issue in Pylons, which does not support debugging in multi-process environments, as the error message indicates.

[This HOWTO](http://hoos.spadial.com/ckan/sysadmin/2017/01/10/howto-start-stop-debug-ckan.html) claims to allow debugging by using `paster serve` instead of httpd. This seems to "work" by the way, except that our server has its own, different (but equally important), breakage in that configuration.

Otherwise, Apache can simply be configured for a single process. Unfortunately, [the _Create the Apache config file_ section of CKAN's maintainer's guide](https://docs.ckan.org/en/latest/maintaining/installing/deployment.html#create-the-apache-config-file) instructs to configure a multi-process CKAN. To workaround, the line `WSGIDaemonProcess ckan_default display-name=ckan_default processes=2 threads=15` must be changed to `WSGIDaemonProcess ckan_default display-name=ckan_default processes=2 threads=15`. This _removes_ the _processes_ option, which - as [the directive's specification](https://modwsgi.readthedocs.io/en/develop/configuration-directives/WSGIDaemonProcess.html) indicates - is _not_ equivalent to setting the value to 1.

References:

- https://helpful.knobs-dials.com/index.php/Mod_wsgi_notes#Process.2Fdaemon_control
- https://www.manongdao.com/q-299749.html

--- a/doc/maintaining/configuration.rst
+++ b/doc/maintaining/configuration.rst
@@ -108,6 +108,15 @@ This enables Pylons' interactive debugging tool, makes Webassets serve
 unminified JS and CSS files, and enables CKAN templates' debugging
 features.
 
+If you are running CKAN on Apache, you must change the WSGI
+configuration to run on single process environment. otherwise
+the execution will fail to AssertionError: The EvalException
+middleware is not usable in a multi-process environment. Eg. change::
+
+  WSGIDaemonProcess ckan_default display-name=ckan_default processes=2 threads=15
+  to
+  WSGIDaemonProcess ckan_default display-name=ckan_default threads=15
+
 .. warning:: This option should be set to ``False`` for a public site.
    With debug mode enabled, a visitor to your site could execute malicious
    commands.

